xarray/core/iterators.py

```diff
@@ -11,9 +14,9 @@ class AbstractIter(abc.Iterator):
     def __init__(
```

Comment:
Considering this `__init__`, using `@dataclass` could be an option here.

Reply:
@Illviljan - sorry for not replying to this comment before. In the latest PR, the `AbstractIter` class was removed in favour or importing directly from `anytree`, so I think this comment is now addressed. Let me know if not, though.

Reply:
If you want, `LevelOrderIter` can still use `@dataclass`.

Reply:
I wasn't familiar with this decorator prior to this PR, but it looks like it generates some magic methods for the class. I'm a little wary of adding them (for example, do we want `LevelOrderIter.__eq__`?). Is there a strong argument here for this class needing them?